DOCSUP-5841: Edit and translate to Russian (#21470)
ClickHouse/ClickHouse: docs/en/sql-reference/table-functions/s3.md

toc_title: s3
---

# S3 Table Function {#s3-table-function}

Provides table-like interface to select/insert files in [Amazon S3](https://aws.amazon.com/s3/). This table function is similar to [hdfs](../../sql-reference/table-functions/hdfs.md), but provides S3-specific features.

**Syntax**

``` sql
s3(path, [aws_access_key_id, aws_secret_access_key,] format, structure, [compression])
```

**Arguments**

> -   `path` — Bucket url with path to file. Supports following wildcards in readonly mode: `*, ?, {abc,def} and {N..M}` where `N, M` — numbers, `'abc', 'def'` — strings.

Reviewer: ```suggestion
-   `path` — Bucket url with path to file. Supports following wildcards in readonly mode: `*`, `?`, `{abc,def}` and `{N..M}` where `N`, `M` — numbers, `'abc'`, `'def'` — strings.
```

A table with the specified structure for reading or writing data in the specified file.

**Examples**

> Table from S3 file `https://storage.yandexcloud.net/my-test-bucket-768/data.csv` and selection of the first two rows from it:

Reviewer: ```suggestion
Selecting the first two rows from the table from S3 file `https://storage.yandexcloud.net/my-test-bucket-768/data.csv`:
```

└─────────┘
```

> 2.  Query the amount of rows in all files of these two directories:

Reviewer: ```suggestion
Count the amount of rows in all files in these two directories:
```

└─────────┘
```

!!! warning "Warning"
    If your listing of files contains number ranges with leading zeros, use the construction with braces for each digit separately or use `?`.

> 3. Query the data from files named `file-000.csv`, `file-001.csv`, … , `file-999.csv`:

Reviewer: ```suggestion
Count the total amount of rows in files named `file-000.csv`, `file-001.csv`, … , `file-999.csv`:
```

└─────────┘
```

> 4. Insert a data into file `test-data.csv.gz`:

Reviewer: ```suggestion
Insert data into file `test-data.csv.gz`:
```

``` sql
INSERT INTO s3('https://storage.yandexcloud.net/my-test-bucket-768/test-data.csv.gz', 'CSV', 'name String, value UInt32', 'gzip')
VALUES ('test-data', 1), ('test-data-2', 2);
```

> 5. Insert a data into file `test-data.csv.gz` from existing table:

Reviewer: ```suggestion
Insert data into file `test-data.csv.gz` from existing table:
```
